src/IO/LasLoader/lasloader.cpp

```diff
@@ -0,0 +1,244 @@
+
+    if ( !fileData->isInitialized() ) {
+        delete fileData;
+        LOG( logWARNING ) << "Filedata could not be initialized";
+        return nullptr;
+    }
+
+    auto startTime = clock();
+    // a unique name is required by the component messaging system
+    static int nameId = 0;
+    auto geometry =
+        make_unique<GeometryData>( "LAS_PC_" + to_string( ++nameId ), GeometryData::POINT_CLOUD );
+    if ( geometry == nullptr ) {
+        LOG( logERROR ) << "could not create geometry";
+        return nullptr;
+    }
+    geometry->setFrame( Ra::Core::Transform::Identity() );
+
+    ifstream stream( filename, ios::out | ios::binary );
+
+    if ( !stream.is_open() ) {
+        throw runtime_error( "Could not open binary ifstream to path " + filename );
+        return nullptr;
+    }
+    else {
+        LOG( logINFO ) << "---Beginning file loading---";
+    }
+
+    /*header part*/
+    char buffer[48];
+
+    // reading minor version (needed to check data format)
+    stream.seekg( 0 );
+    stream.seekg( 25 );
+    stream.read( buffer, 1 );
+    unsigned char minor = *(unsigned char*)buffer;
+
+    if ( ( minor < 1 ) || ( minor > 4 ) ) {
+        delete fileData;
+        LOG( logERROR ) << "LAS version 1." << (int)minor << " unsupported";
+        return nullptr;
+    }
+    else {
+        LOG( logINFO ) << "LAS version 1." << (int)minor;
+    }
+
+    // reading distance from file start to first point record
+    stream.seekg( 0 );
+    stream.seekg( 96 );
+    stream.read( buffer, 4 );
+    unsigned int offset = *(unsigned int*)buffer;
+
+    // reading point data format
+    stream.seekg( 0 );
+    stream.seekg( 104 );
+    stream.read( buffer, 1 );
+    unsigned char data_format = *(unsigned char*)buffer;
+
+    if ( ( data_format < 0 ) || ( data_format > 3 && minor < 3 ) ||
+         ( data_format > 5 && minor == 3 ) || ( data_format > 6 && minor == 4 ) ) {
+        delete fileData;
+        throw runtime_error( "Corrupted file. Unvalid data format" );
+        return nullptr;
+    }
+    else {
+        LOG( logINFO ) << "Data format " << (int)data_format;
+        LOG( logINFO ) << "Loading properties \"x\", \"y\", \"z\",";
+    }
+
+    // reading data record length (bytes)
+    stream.read( buffer, 2 );
+    unsigned short data_len = *(unsigned short*)buffer;
+
+    // reading total number of data records
+    stream.read( buffer, 4 );
+    unsigned int nb_data = *(unsigned int*)buffer;
+
+    // reading scales and offsets (used for computing coordinates)
+    stream.seekg( 0 );
+    stream.seekg( 131 );
+    stream.read( buffer, 48 );
+
+    double scale_x  = *(double*)buffer;
+    double scale_y  = *(double*)( buffer + 8 );
+    double scale_z  = *(double*)( buffer + 16 );
+    double offset_x = *(double*)( buffer + 24 );
+    double offset_y = *(double*)( buffer + 32 );
+    double offset_z = *(double*)( buffer + 40 );
+
+    /*loading properties*/
+    char* point = (char*)malloc( data_len * sizeof( char ) );
```

Comment:
Better use `std::vector` with `reserve` or `resize` to avoid multiple (re)allocations.

Reply:
Access to raw buffet (after `resize`) using the method `data()`.
https://en.cppreference.com/w/cpp/container/vector/data

Reply:
done